### PR TITLE
support filter in test performance

### PR DESCRIPTION
**Pull request by inabao**

### CODE REVIEW

File: tests/performance/test_performance.cpp [hunk 1]
```diff
@@ -44,7 +44,6 @@ const static std::string META_DATA_FILE = "_meta.data";
 
 int
 main(int argc, char* argv[]) {
-    set_level(level::off);
```
Comment (wxyucs) [inline tests/performance/test_performance.cpp]: what is the reason to remove this line?

File: tests/performance/test_performance.cpp [hunk 2]
```diff
@@ -349,7 +372,7 @@ class Test {
         // search
         auto search_start = std::chrono::steady_clock::now();
         int64_t correct = 0;
-        int64_t total = test_dataset->GetNumberOfQuery();
+        int64_t total = 100;
```
Comment (jiaweizone) [inline tests/performance/test_performance.cpp]: Why change to constant value `100` ?

Comment (inabao) [reply tests/performance/test_performance.cpp]: i has restored it.

File: tests/performance/test_performance.cpp (hunk 1 above)
Comment (inabao) [reply tests/performance/test_performance.cpp]: i has restored it.

File: tests/performance/test_performance.cpp [hunk 3]
```diff
@@ -176,6 +178,21 @@ class TestDataset {
...
+            dataset.read(obj->train_labels_.get(), datatype, dataspace);
+        }
+
+        if (has_labels) {
+            H5::DataSet dataset = file.openDataSet("/test_labels");
+            H5::DataSpace dataspace = dataset.getSpace();
+            H5::FloatType datatype(H5::PredType::NATIVE_INT64);
+            obj->test_labels_ = std::shared_ptr<int64_t[]>(new int64_t[obj->number_of_query_]);
+            dataset.read(obj->test_labels_.get(), datatype, dataspace);
+        }
```
Comment (wxyucs) [inline tests/performance/test_performance.cpp]: add comments about why separate code into two scopes

**Review by wxyucs — CHANGES_REQUESTED** [review]

File: tests/performance/test_performance.cpp (hunk 3 above)
Comment (inabao) [reply tests/performance/test_performance.cpp]: revised

**Review by wxyucs — APPROVED** [review]
lgtm

**Review by jiaweizone — APPROVED** [review]
LGTM

**Pull request merged**